server/src/controller/categoriesQuery.js

```diff
@@ -0,0 +1,32 @@
+import validate from "../util/checkValidation.js";
+import prisma from "../../prisma/client.js";
+import { StatusCodes } from "http-status-codes";
+
+/**
+ * Is virtually the same as categoryController.js getCategory()
+ * Task: query: list of categories in the warehouse #38
+ */
+export const getCategoriesInWarehouse = async (req, res) => {
+  if (validate(req, res)) {
+    return res;
+  }
+
+  const [pallots, categories] = await Promise.all([prisma.pallot.findMany(), prisma.category.findMany()]);
+
+  let categoryIds_set = new Set(); // no duplicates
+  for (const pallot of pallots) {
+    for (const categoryIds of pallot.categoryIds) {
+      for (const categoryId of categoryIds) {
```

Comment:
also you have an extra for loop here

Reply:
I believe that categoryIds is of type Int[], so I think the for loop is needed? I might not be understanding this correctly, pls lmk

<img width="697" alt="Screen Shot 2023-03-03 at 1 01 20 PM" src="https://user-images.githubusercontent.com/87678394/222794147-a7811eee-9e91-4bab-83b7-8efcaad545f5.png">

Reply:
Are you saying that each "pallot" has only one "categoryIds"?

Reply:
I think I understand now

Reply:
used flatMap() btw for better readability